dandiapi/api/models/asset.py

```diff
@@ -105,8 +109,16 @@ class Status(models.TextChoices):
         VALID = 'Valid'
         INVALID = 'Invalid'
 
+    def validate_path(path: str):
+        if path.startswith('/'):
+            raise ValidationError('Path must not begin with /')
+        if not re.match(ASSET_PATH_REGEX, path):
+            raise ValidationError(f'Path violates regex: {ASSET_PATH_REGEX}')
```

Comment:
I think we should avoid putting regular expressions into our error messages. Something like "Path is improperly formatted" is probably fine.

Reply:
My concern was that someone might not know what constitutes a valid asset path, and so wouldn't know how to fix their error. Any suggestions regarding that?

Reply:
I agree it's a problem, but given the level of dependency on hard coded error messages right now - I don't want to create an error message with a very-likely-to-change-in-the-future regex. I think the solution is general documentation, since end users are often not going to able to decipher a regex-laden error message anyway. Either way I don't think there's any blocker to this PR right now, I would just make the error message more generic.

Reply:
For error messages such as this (where the fix is in scope for a CS person, but perhaps not for someone in a different field) we can also put a link in the error message encouraging them to get in touch via the helpdesk.

Reply:
[`312d74d` (#1312)](https://github.com/dandi/dandi-archive/pull/1312/commits/312d74d45372619678d0a84ffdac211fc2b50bb2)